agenta-cli/agenta/sdk/agenta.py

```diff
@@ -151,46 +157,85 @@ def wrapper(*args, **kwargs):
             return result
         except Exception as e:
             if sys.version_info.major == 3 and sys.version_info.minor < 10:
-                traceback_str = ''.join(traceback.format_exception(None, e, e.__traceback__))
+                traceback_str = "".join(
+                    traceback.format_exception(None, e, e.__traceback__)
+                )
             else:
-                traceback_str = ''.join(traceback.format_exception(e, value=e, tb=e.__traceback__))
-            return JSONResponse(status_code=500, content={"error": str(e), "traceback": traceback_str})
+                traceback_str = "".join(
+                    traceback.format_exception(e, value=e, tb=e.__traceback__)
+                )
+            return JSONResponse(
+                status_code=500,
+                content={"error": str(e), "traceback": traceback_str},
+            )
+
     new_params = []
+    instances_to_override = []
     for name, param in sig.parameters.items():
         if name in app_params:
-            new_params.append(
-                inspect.Parameter(
-                    name,
-                    inspect.Parameter.KEYWORD_ONLY,
-                    default=Body(app_params[name]),
-                    annotation=Optional[param.annotation]
+            if param.annotation is MultipleChoiceParam:
+                # Append the instance and its name to the list
+                instances_to_override.append((name, app_params[name]))
+                new_params.append(
+                    inspect.Parameter(
+                        name,
+                        inspect.Parameter.KEYWORD_ONLY,
+                        default=Body(app_params[name]),
+                        annotation=Optional[param.annotation],
+                    )
+                )
+            else:
+                new_params.append(
+                    inspect.Parameter(
+                        name,
+                        inspect.Parameter.KEYWORD_ONLY,
+                        default=Body(app_params[name]),
+                        annotation=Optional[param.annotation],
+                    )
                 )
-            )
         else:
             new_params.append(
                 inspect.Parameter(
                     name,
                     inspect.Parameter.KEYWORD_ONLY,
                     default=Body(...),
-                    annotation=param.annotation
-
+                    annotation=param.annotation,
                 )
             )
 
     wrapper.__signature__ = sig.replace(parameters=new_params)
 
     route = "/generate"
+    func_name = func.__name__
```

Comment:
I think that's wrong! The schema will be found in "Body_generate... and not in func.__name__ since we are not using the func name for the api endpoint

Reply:
We are. Presently, I'm testing with the sales_transcript_summarizer example, and its function name is `query`.

Run the serve command and cancel it when the agenta folder has been regenerated. Head to the SDK -> `agenta/agenta.py`, and scroll down to line 215. In the decorator function, post, where you see: 

```python 
schema = app.openapi()  # or app.openapi_schema
```  

Include this print (```python print("Schemas: ", schema["components"]["schemas"])````) statement after the above schema line and try accessing the app from the API docs.

In your console, you'll see:

```bash
Schemas:  {'Body_query_generate_post': {'title': 'Body_query_generate_post', ...}}
```
If you rename your app function from `query` to `generate`, the schema gets updated to this:

```bash
Schemas:  {'Body_query_generate_post': {'title': 'Body_query_generate_post', ...}}
```

Reply:
We are. Presently, I'm testing with the `sales_transcript_summarizer` example, and its function name is `query`.

Run the serve command and cancel it when the agenta folder has been regenerated. Head to the SDK -> `agenta/agenta.py`, and scroll down to line 215. In the decorator function, post, where you see: 

```python 
schema = app.openapi()  # or app.openapi_schema
```  

Include this `print("Schemas: ", schema["components"]["schemas"])` after the above schema line and try accessing the app from the API docs.

In your console, you'll see:

```bash
Schemas:  {'Body_query_generate_post': {'title': 'Body_query_generate_post', ...}}
```
If you rename your app function from `query` to `generate`, the schema gets updated to this:

```bash
Schemas:  {'Body_generate_generate_post': {'title': 'Body_generate_generate_post', ...}}
```